plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSourceProvider.java

```diff
@@ -146,12 +157,12 @@ public ConnectorPageSource createPageSource(
             List<ColumnHandle> columns,
             DynamicFilter dynamicFilter)
     {
-        HudiSplit split = (HudiSplit) connectorSplit;
-        Path path = new Path(split.getPath());
-        HoodieFileFormat hudiFileFormat = getHudiFileFormat(path.toString());
-        if (!HoodieFileFormat.PARQUET.equals(hudiFileFormat)) {
-            throw new TrinoException(HUDI_UNSUPPORTED_FILE_FORMAT, format("File format %s not supported", hudiFileFormat));
-        }
+        HudiTableHandle tableHandle = (HudiTableHandle) connectorTable;
+        HudiTableType tableType = tableHandle.getTableType();
+        HudiSplit hudiSplit = (HudiSplit) connectorSplit;
+        final HudiFile baseFile = hudiSplit.getBaseFile().orElseThrow(() ->
+                new TrinoException(HUDI_CANNOT_OPEN_SPLIT, "Split without base file is invalid"));
```

Comment:
Actually, Hudi now also supports writing only to log files for MOR tables. Maybe, that can be taken up as a follow-up. But, since we already upgraded to 0.12.1, it would be great if we can do it in this PR itself.

Reply:
Hi codope
Thanks for your reminder, I will research it. Not sure how long will it take to implement writing to log files for MOR tables. 
So, maybe we can raise another PR for this.

Reply:
> Hi codope Thanks for your reminder, I will research it. Not sure how long will it take to implement writing to log files for MOR tables. So, maybe we can raise another PR for this.

@willzgw I feel what codope means is there should be some HudiSplits that only have logs but no base file, can we support reading a FileGroup without base files?

Reply:
Thank you guys, I've got this done, please help review.